### ISSUE TITLE
unable to build 2gp package with the latest cli due to big object index file

### ISSUE THREAD

**Issue by bill-meyestro** [forcedotcom/cli#2847]
> **Note** 
> Before you submit your issue, make sure that:
> - You're using the latest version of Salesforce CLI.
> - You've searched both open and closed issues for related posts.
> - You've used the `doctor` command to diagnose common issues. 
> - You understand that GitHub Issues don't adhere to any agreement or SLA.
>   - If you require immediate assistance, use official channels such as Salesforce Customer Support.

### Summary

I upgraded my cli from 2.35.6 to 2.38.7 this morning and tried to build my project (2gp). Getting the following error -

Audit__b.AuditIndex.indexe: An object 'Audit__b.AuditIndex.indexe' of type Index was named in package.xml, but was not found in zipped directory

This is not a new object.

I believe it was a couple of years ago we had to rename the big object index file to ".indexe-meta.xml" as a workaround in the build or deploy process.

Does the file need to be renamed to "index-meta.xml" again?

### System Information

{
  "architecture": "win32-x64",
  "cliVersion": "@salesforce/cli/2.38.7",
  "nodeVersion": "node-v20.12.2",
  "osVersion": "Windows_NT 10.0.22631",
  "rootPath": "C:\\Users\\BillDunn\\AppData\\Local\\sf\\client\\2.38.7-023d2b4",
  "shell": "cmd.exe",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.15 (core)",
    "@oclif/plugin-commands 3.3.1 (core)",
    "@oclif/plugin-help 6.0.21 (core)",
    "@oclif/plugin-not-found 3.1.4 (core)",
    "@oclif/plugin-plugins 5.0.10 (core)",
    "@oclif/plugin-search 1.0.22 (core)",
    "@oclif/plugin-update 4.2.6 (core)",
    "@oclif/plugin-version 2.0.17 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.15 (core)",
    "@oclif/plugin-which 3.1.7 (core)",
    "@salesforce/cli 2.38.7 (core)",
    "apex 3.1.5 (core)",
    "auth 3.6.1 (core)",
    "data 3.3.1 (core)",
    "deploy-retrieve 3.6.3 (core)",
    "info 3.2.1 (core)",
    "limits 3.3.3 (core)",
    "marketplace 1.2.1 (core)",
    "org 4.1.1 (core)",
    "packaging 2.4.0 (core)",
    "schema 3.3.3 (core)",
    "settings 2.2.1 (core)",
    "sobject 1.3.3 (core)",
    "source 3.3.1 (core)",
    "telemetry 3.3.2 (core)",
    "templates 56.2.2 (core)",
    "trust 3.6.3 (core)",
    "user 3.5.2 (core)",
    "@salesforce/sfdx-scanner 4.0.0 (user)"
  ]
}

**Comment by mshanemc** [comment]
`indexe` is definitely deprecated and you shouldn't use it.  But that hasn't changed recently.

what command are you running when you get the error?  `sf package version create` ?

**Comment by bill-meyestro** [comment]
sf package version create --package "xxxxx" --installation-key zzzzzzzzz --definition-file config/project-scratch-def.json --code-coverage

**Comment by bill-meyestro** [comment]
i am rebuilding with "index-meta.xml" and will deploy to a scratch org and upgrade a customer sandbox.

**Comment by bill-meyestro** [comment]
No trouble building, deploying to scratch org, or upgrading the package in a sandbox. Thank you very much for such a quick response. I really appreciate it.

**Comment by jongpie** [comment]
@mshanemc I've also just [upgraded to sf cli `v2.38.7`](https://github.com/jongpie/NebulaLogger/commit/8e6730e8fec1f287798e5c088c2a94cda1ba83bd#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R83), and I'm now seeing this same error:

- For me, it's happening [when I run `sf project deploy start`](https://github.com/jongpie/NebulaLogger/actions/runs/8854153222/job/24316694777#step:7:1) 
- The directory being deployed has [1 Big Object](https://github.com/jongpie/NebulaLogger/tree/8e6730e8fec1f287798e5c088c2a94cda1ba83bd/nebula-logger/plugins/big-object-archiving/plugin/objects/LogEntryArchive__b)
- It fails to deploy due to the same type of error, [`An object 'LogEntryArchive__b.LogEntryArchiveIndex.indexe' of type Index was named in package.xml, but was not found in zipped directory`](https://github.com/jongpie/NebulaLogger/actions/runs/8854153222/job/24316694777#step:7:113)
- The error message says it's looking for an `.indexe` file, but I'm using [an `.indexe-meta.xml` file](https://github.com/jongpie/NebulaLogger/blob/8e6730e8fec1f287798e5c088c2a94cda1ba83bd/nebula-logger/plugins/big-object-archiving/plugin/objects/LogEntryArchive__b/indexes/LogEntryArchiveIndex.indexe-meta.xml)
- None of the big object's metadata has been changed in a while
- This seems to be happening for me both locally on Windows 11 and in GitHub Actions [running `ubuntu-latest`](https://github.com/jongpie/NebulaLogger/blob/8e6730e8fec1f287798e5c088c2a94cda1ba83bd/.github/workflows/build.yml#L130)

I haven't tried to create a package version yet (I can test that later), but it seems to be happening every time I deploy/push to scratch orgs.

Let me know what you think - happy to provide any other info that might be helpful.

**Comment by mshanemc** [comment]
@jongpie short answer **and workaround**: you should use `index-meta.xml` instead of `indexe`.  Does it work if you take your CLI back to whatever you were on before?  If so, what version was that?
 
I'm going to look into whether something changed recently related to that.

---

background: `indexe` was a typo from the old toolbelt days, and we put in a shim to support the use of it for backward compatibility reasons.  
https://github.com/forcedotcom/source-deploy-retrieve/blob/1e93fd9991c28acd5db41d451b866edc29e660ca/src/registry/metadataRegistry.json#L1495

That's used here https://github.com/forcedotcom/source-deploy-retrieve/blob/1e93fd9991c28acd5db41d451b866edc29e660ca/src/convert/transformers/defaultMetadataTransformer.ts#L78

[still trying to figure out why it's not working like it used to!]

**Comment by jongpie** [comment]
@mshanemc I reverted to the previous version I was using - `v2.24.4` - and was able to deploy successfully using the legacy `indexe-meta.xml` extension. So it seems like something has changed between `v2.24.4` and `v2.38.7`.

I've also switched to using the new/correct `index-meta.xml` extension, and that works for me with `v2.38.7`, so I'm not blocked by this anymore - but if there's anything else I can do to help troubleshoot why `indexe` stopped working between versions, let me know.
